client/my-sites/checkout/composite-checkout/lib/submit-wpcom-transaction.ts

```diff
@@ -17,8 +17,15 @@ export default async function submitWpcomTransaction(
 	payload: WPCOMTransactionEndpointRequestPayload,
 	transactionOptions: PaymentProcessorOptions
 ): Promise< WPCOMTransactionEndpointResponse > {
-	if ( transactionOptions && transactionOptions.createUserAndSiteBeforeTransaction ) {
-		return createAccount().then( ( response ) => {
+	if (
+		( transactionOptions && transactionOptions.createUserAndSiteBeforeTransaction ) ||
```

Comment:
Minor (and not something you had changed), but if we can trust TS, then `transactionOptions` is always set here, so we could probably remove the extra check for its existence and simplify this condition slightly.